### PR TITLE
Check if loaded before delegator search/query

### PR DESCRIPTION
**Pull request by bigsheeper**
issue: #23141

### CODE REVIEW

**Comment by bigsheeper** [comment]
rerun ut

File: internal/querynodev2/delegator/delegator.go [hunk 1]
```diff
@@ -205,6 +205,12 @@ func (sd *shardDelegator) Search(ctx context.Context, req *querypb.SearchRequest
 		return nil, fmt.Errorf("dml channel not match, delegator channel %s, search channels %v", sd.vchannelName, req.GetDmlChannels())
 	}
 
+	partitions := req.GetReq().GetPartitionIDs()
+	if !sd.collection.ExistPartition(partitions...) {
+		return nil, fmt.Errorf("partitions not loaded into memory, collectionID=%d, partitionIDs=%s",
```
Comment (yah01) [inline internal/querynodev2/delegator/delegator.go:210]: use `merr.WrapPartitionNotLoaded`

File: internal/querynodev2/delegator/delegator.go [hunk 2]
```diff
@@ -251,6 +257,12 @@ func (sd *shardDelegator) Query(ctx context.Context, req *querypb.QueryRequest)
...
```
Comment (yah01) [inline internal/querynodev2/delegator/delegator.go:262]: Same as line 210

**Review by yah01 — CHANGES_REQUESTED** [review]

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**